packages/smooth_app/lib/data_models/food_preferences_model.dart

```diff
@@ -0,0 +1,209 @@
+  final String id;
+  final String name;
+  final String default_value;
+  final String icon_url;
+  final String setting_name;
+
+  Attribute({this.id, this.name, this.default_value, this.icon_url, this.setting_name});
+
+  factory Attribute.fromJson(Map<String, dynamic> json) {
+
+    log("Attribute factory", error: json);
+
+    return new Attribute(
+      id: json['id'] as String,
+      name: json['name'] as String,
+      default_value: json['default'] as String,
+      icon_url: json['icon_url'] as String,
+      setting_name: json['setting_name'] as String,
+    );
+  }
+}
+
+class AttributeGroup {
+  final String id;
+  final String name;
+  final List<Attribute> attributes;
+
+  AttributeGroup({this.id, this.name, this.attributes});
+
+  factory AttributeGroup.fromJson(Map<String, dynamic> json) {
+
+    log("AttributeGroup factory");
+
+    var attributesList = json['attributes'] as List;
+
+    List<Attribute> attributes = new List<Attribute>();
+    attributes = attributesList.map((i)=>Attribute.fromJson(i)).toList();
+
+    return new AttributeGroup(
+      id: json['id'] as String,
+      name: json['name'] as String,
+      attributes: attributes,
+    );
+  }
+}
+
+class AttributeGroups {
+  // ignore: non_constant_identifier_names
+  final List<AttributeGroup> attribute_groups;
+
+  AttributeGroups({
+    this.attribute_groups,
+  });
+
+  factory AttributeGroups.fromJson(List<dynamic> json) {
+
+    log("AttributeGroups factory");
+
+    List<AttributeGroup> attribute_groups = new List<AttributeGroup>();
+    attribute_groups = json.map((i)=>AttributeGroup.fromJson(i)).toList();
+
+    return new AttributeGroups(
+      attribute_groups: attribute_groups,
+    );
+  }
+}
+
+Future<AttributeGroups> fetchAttributeGroups(http.Client client) async {
+  final response =
+  await client.get('https://world.openfoodfacts.org/api/v0/attribute_groups');
+
+  final json = jsonDecode(response.body).cast<Map<String, dynamic>>();
+
+  return(AttributeGroups.fromJson(json));
+}
+
+class FoodPreferencesModel extends ChangeNotifier {
+  FoodPreferencesModel() {
+    userDatabase = UserDatabase();
+    _loadData();
+  }
+
+  Future<bool> _loadData() async {
+    try {
+      preferences = await fetchPreferences(http.Client());
+      userPreferences = await userDatabase.getUserPreferences();
+      attributeGroups = await fetchAttributeGroups(http.Client());
+      dataLoaded = true;
+      notifyListeners();
+      return true;
+    } catch (e) {
+      print('An error occurred while loading user preferences : $e');
+      dataLoaded = false;
+      return false;
+    }
+  }
+
+  List<Preference> preferences;
+  UserDatabase userDatabase;
+  UserPreferences userPreferences;
+  bool dataLoaded = false;
+  AttributeGroups attributeGroups;
+
+  UserPreferencesVariableValue getVariable(UserPreferencesVariable variable) {
+    return userPreferences.getVariable(variable);
+  }
+
+  void setVariable(UserPreferencesVariable variable, int value) {
```

Comment:
To me we cannot have `_vegan` fields in user preferences anymore if we want flexibility, we need rather something like `_values['vegan']`. And as you don't change `user_preferences.dart` (and `UserPreferencesVariable`) I don't see how you can do that.

Reply:
Agreed.